src/main/util/conversions.cc

```diff
@@ -1518,38 +1524,48 @@ int recordbins_from_jsobject(as_record *rec, Local<Object> obj,
 		const Local<Value> value = Nan::Get(obj, name).ToLocalChecked();
 
 		Nan::Utf8String n(name);
-		if (strlen(*n) > AS_BIN_NAME_MAX_SIZE) {
-			as_v8_error(log, "Bin name length exceeded (max. %i): %s",
-						AS_BIN_NAME_MAX_SIZE, *n);
-			return AS_NODE_PARAM_ERR;
-		}
-
+		
 		if (value->IsUndefined()) {
 			as_v8_error(log, "Bin value 'undefined' not supported: %s", *n);
 			return AS_NODE_PARAM_ERR;
 		}
 		if (value->IsNull()) {
-			as_record_set_nil(rec, *n);
+			bool success = as_record_set_nil(rec, *n);
+			if (!success) {
+				return throw_bin_name_error(*n, log);
```

Comment:
`as_record_set_nil(rec, *n);` might fail for reasons unrelated to bin name error